packages/medusa/src/api/routes/store/products/get-product.ts

```diff
@@ -89,13 +104,27 @@ export default async (req, res) => {
     currencyCode = region.currency_code
   }
 
-  const [product] = await pricingService.setProductPrices([rawProduct], {
-    cart_id: validated.cart_id,
-    customer_id: customer_id,
-    region_id: regionId,
-    currency_code: currencyCode,
-    include_discount_prices: true,
-  })
+  const pricedProductArray = await pricingService.setProductPrices(
+    [rawProduct],
+    {
+      cart_id: validated.cart_id,
+      customer_id: customer_id,
+      region_id: regionId,
+      currency_code: currencyCode,
+      include_discount_prices: true,
+    }
+  )
+
+  const [product] = await productVariantInventoryService.setProductAvailability(
+    pricedProductArray,
+    sales_channel_id
+  )
 
   res.json({ product })
 }
+
+export class StoreGetProductParams extends PriceSelectionParams {
+  @IsString()
+  @IsOptional()
+  sales_channel_id?: string
+}
```

Comment:
todo: add middleware to the endpoint that checks if the PublishableKey has access to the provided `sales_channel_id`